src/docfx/template/TemplateUtility.cs

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) Microsoft. All rights reserved.
```

Comment:
why do we separate these methods here? they are only used in `TemplateEngine`, just because that file is too long?